app/global_invite/rest.py

```diff
@@ -50,3 +50,15 @@ def validate_invitation_token(invitation_type, token):
         return jsonify(data=invited_user.serialize()), 200
     else:
         raise InvalidRequest("Unrecognised invitation type: {}".format(invitation_type))
+
+
+@global_invite_blueprint.route('/service/<uuid:invited_user_id>', methods=['GET'])
+def get_invited_user(invited_user_id):
+    invited_user = get_invited_user_by_id(invited_user_id)
+    return jsonify(data=invited_user_schema.dump(invited_user).data), 200
+
+
+@global_invite_blueprint.route('/organisation/<uuid:invited_org_user_id>', methods=['GET'])
+def get_invited_org_user(invited_org_user_id):
+    invited_user = dao_get_invited_organisation_user(invited_org_user_id)
+    return jsonify(data=invited_user.serialize()), 200
```

Comment:
Note to self: the different use of `serialize` vs `dump` here appears to be historical.

- The service user invite schema adds an `auth_type` foreign key to the dumped data (see [here](https://github.com/alphagov/notifications-api/blob/4cbd4c092090739ebfefc9d2b906bf8b39b213e4/app/schemas.py#L624)) and otherwise dumps all columns.
- The org user invite defines its own `serialize` method to specify the columns to dump in a dictionary.

Both perform validation of the email address - the org user invite has a specific schema for the posted data to achieve this (see [here](https://github.com/alphagov/notifications-api/blob/4cbd4c092090739ebfefc9d2b906bf8b39b213e4/app/organisation/invite_rest.py#L34)).

It looks like we've taken two approaches:

- Some endpoints have dedicated schemas, and a `serialize` method on the model.
- Other endpoints share a schema, which is reused for serialization.

@leohemsted just to help me understand a bit better: do you have any thoughts on which one of these approaches we prefer to use?

Reply:
hey, good question, there's a bit of legacy here.

the stuff in `schemas.py` is marshmallow, which we don't love (we're on an old version cos new version breaks, it's full of mystery and magic and surprises) and are slowly (glacially) moving away from that (and the `dump`) method towards `serialize` methods on the models when returning, and using json schemas to validate input data (note that we still need to do the deserialization ourselves, eg to datetimes or from str to UUID etc).

Reply:
Thanks for explaining, that's good to know. Sounds like the new way has some advantages, especially as the load vs. dump are less coupled (and we're not stuck on some ancient version of something).

Do you think it's worth documenting this [here](https://github.com/alphagov/notifications-manuals/wiki/Decision-records)?

Reply:
yeah, almost certainly. i'll write something up later